src/server/pfs/server/obj_block_api_server.go

```diff
@@ -265,30 +271,11 @@ func (s *objBlockAPIServer) putObject(ctx context.Context, dataReader io.Reader,
 				retErr = err
 			}
 		}()
-		if split {
-			size, err = io.CopyN(w, r, pfsclient.ChunkSize)
-		} else {
-			buf := grpcutil.GetBuffer()
-			defer grpcutil.PutBuffer(buf)
-			size, err = io.CopyBuffer(w, r, buf)
-		}
-		if err != nil {
-			if err != io.EOF {
-				s.objClient.Delete(ctx, blockPath)
-			}
-			return err
-		}
-		return nil
+		size, err = f(w, r)
+		return err
 	}(); err != nil {
-		if err == io.EOF {
-			defer func() {
-				if retErr == nil {
-					retErr = io.EOF
-				}
-			}()
-		} else {
-			return nil, err
-		}
+		s.objClient.Delete(ctx, s.blockPath(block))
```

Comment:
This returns an error right? We're probably ok swallowing that error since if we can't delete the file what are we going to do, but given our recent mishaps with swallowing errors it's probably a good idea to comment that this is intentional and why it's reasonable.

Reply:
Makes sense.